hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -771,6 +771,9 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 		}
 		validReleaseImage := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.ValidReleaseImage))
 		if validReleaseImage != nil && validReleaseImage.Status == metav1.ConditionFalse {
+			if strings.Contains(validReleaseImage.Message, "failed to get pull secret") {
```

Comment:
this should be checking the "reason" which is for programatic purposes. "Message" is meant for human consumption

Reply:
👍 Updated